pkg/agent/controller/egress/egress_controller.go

```diff
@@ -613,29 +614,71 @@ func (c *EgressController) unbindPodEgress(pod, egress string) (string, bool) {
 	return "", false
 }
 
-func (c *EgressController) updateEgressStatus(egress *crdv1b1.Egress, egressIP string) error {
+func (c *EgressController) updateEgressStatus(egress *crdv1b1.Egress, egressIP string, scheduleErr error) error {
 	isLocal := false
 	if egressIP != "" {
 		isLocal = c.localIPDetector.IsLocalIP(egressIP)
 	}
+
+	desiredStatus := &crdv1b1.EgressStatus{}
+	if isLocal {
+		desiredStatus.EgressNode = c.nodeName
+		desiredStatus.EgressIP = egressIP
+		if isEgressSchedulable(egress) {
+			desiredStatus.Conditions = []crdv1b1.EgressCondition{
+				{
+					Type:               crdv1b1.IPAssigned,
+					Status:             v1.ConditionTrue,
+					LastTransitionTime: metav1.Now(),
+					Reason:             "Assigned",
+					Message:            "EgressIP is successfully assigned to EgressNode",
+				},
+			}
+		}
+	} else if egressIP == "" {
+		// Select one Node to update false status among all Nodes.
+		nodeToUpdateStatus, err := c.cluster.SelectNodeForIP(egress.Spec.EgressIP, "")
+		if err != nil {
+			return err
+		}
+		// Skip if the Node is not the selected one.
+		if nodeToUpdateStatus != c.nodeName {
+			return nil
+		}
+		desiredStatus.EgressNode = ""
+		desiredStatus.EgressIP = ""
+		// If the error is nil, it means the Egress hasn't been processed yet. Therefore, we only set IPAssigned
+		// condition to false when there is an error.
+		if scheduleErr != nil {
```

Comment:
The interval will just be a few ms. And there could be some bad side effects, so I intentionally add the check:
1. One intermediate state will be generated and overwriten very soon.
2. Two agents may try to update it and cause some backoff retries, which may delay the update of the right condition. One agent is selected due to fallback mechanism (when the Egress is not scheduled to any node because it's not processed yet), one agent is the right owner (after it's processed).

The state an Egress is not processed will last only a very short time. It's because the scheduler may receive the event at the same time as the controller. So controller may process it before the scheduler gets a result. But as long as the Egress is there, the scheduler will eventually have a result for it very soon.

Also added comments for it.